### PR TITLE
Fy/dev

### PR DESCRIPTION
**Pull request by 2020dfff**
### Summary :memo:
Use 6 metrics to judge MAS topology, draw the plot

### Details
_Describe more what you did on changes._
1. fig_plot, add function to plot scatter, boxplot
2. for topology(100 graphs each) but not within one graph
3. x-axis: Graph ID or Diameter

### CODE REVIEW

File: tpf/fig_plot.py [hunk 1]
```diff
@@ -16,10 +16,21 @@
 class GraphPlotter:
     @classmethod
     def plot_graph(cls, graph: nx.DiGraph, name: str, path: str):
-        plt.figure(figsize=(8, 8))
-        pos = nx.spring_layout(graph)
+        # if name == "chain_xxx": use circular layout
```
Comment (jjkysy) [inline tpf/fig_plot.py]: 这个有点奇怪啊，之后如果用这个库画一些别的智能体，不是用这个命名方式命名的咋办？放在这里有点奇怪。

File: tpf/fig_plot.py [hunk 2]
```diff
@@ -56,10 +78,143 @@ def plot_centrality_features(cls, feature: GraphFeatures, path: str):
...
     @classmethod
-    def plot_features(cls, features: List[GraphFeatures], path: str):
+    def plot_features_within_one_graph(
+        cls, features: List[GraphFeatures], path: str
+    ):
         for feature_set in features[:1]:
             cls.plot_centrality_features(feature_set, path)
+
+    @classmethod
+    def plot_size_features_for_topologies(
```
Comment (jjkysy) [inline tpf/fig_plot.py]: plot这块写的稍微有点丑了哈哈哈。（1）把计算的逻辑、画图的逻辑、存储的逻辑分开。模块化一点，之后别人也好用。（2）套用画图模版，e.g., import seaborn as sn 
sn.set_theme('paper', 'whitegrid')

**Review by jjkysy — CHANGES_REQUESTED** [review]
请修改fig_plot里的内容（1）将名字相关的判断语句和固定的名字剥离，可以创造单独的数据格式，用列表或者字典存储。（2）将计算逻辑、画图逻辑、存储逻辑模块化分开，增加可读和可修改性。（3）增加画图模版seabon

**Comment by 2020dfff** [comment]
- take the storage part out
- seperate graph feature and topo feature
- class method to instance method
- name: store with dict 
- sns set as whitegrid for now

TODO:
Add test

**Comment by 2020dfff** [comment]
Relogic the calculation, from plot module to evaluation module

**Review by jjkysy — COMMENTED** [review]
LGTM

**Comment by jjkysy** [comment]
LGTM

**Pull request merged**